resource/systemd/unit/unit.go

```diff
@@ -0,0 +1,543 @@
+		linked-runtime, masked, masked-runtime, static, disabled, invalid. enabled
+		indicates that a unit file is permanently enabled. enable-runtime indicates
+		the unit file is only temporarily enabled, and will no longer be enabled
+		after a reboot (that means, it is enabled via /run symlinks, rather than /etc).
+		linked indicates that a unit is linked into /etc permanently, linked indicates
+		that a unit is linked into /run temporarily (until the next reboot). masked
+		indicates that the unit file is masked permanently, masked-runtime indicates
+		that it is only temporarily masked in /run, until the next reboot.
+		static indicates that the unit is statically enabled, i.e. always enabled and
+		doesn't need to be enabled explicitly. invalid indicates that it could not
+		be determined whether the unit file is enabled.
+	*/
+	UnitFileState systemd.UnitFileState
+
+	/* Mode for the call to StartUnit()
+	StartUnit() enqeues a start job, and possibly depending jobs.
+	Takes the unit to activate, plus a mode string. The mode needs to be one of
+	replace, fail, isolate, ignore-dependencies, ignore-requirements.
+	If "replace" the call will start the unit and its dependencies,
+	possibly replacing already queued jobs that conflict with this. If "fail" the
+	call will start the unit and its dependencies, but will fail if this would
+	change an already queued job. If "isolate" the call will start the unit in
+	question and terminate all units that aren't dependencies of it. If
+	"ignore-dependencies" it will start a unit but ignore all its dependencies.
+	If "ignore-requirements" it will start a unit but only ignore the requirement
+	dependencies. It is not recommended to make use of the latter two options.
+	Returns the newly created job object.
+	*/
+	StartMode systemd.StartMode // how to start the unit
+
+	// the amount of time the command will wait for configuration to load
+	// before halting forcefully. The
+	// format is Go's duraction string. A duration string is a possibly signed
+	// sequence of decimal numbers, each with optional fraction and a unit
+	// suffix, such as "300ms", "-1.5h" or "2h45m". Valid time units are "ns",
+	// "us" (or "µs"), "ms", "s", "m", "h".
+	Timeout time.Duration // how long to wait for the load state to resolve
+}
+
+// Check if all the properties for the unit are correct
+// 1. Checks if the unit is currently loading, if so waits Default 5 seconds
+// 2. Checks if the unit is in the active state
+// 3. Check if the unit is in the unit file state
+func (t *Unit) Check(r resource.Renderer) (resource.TaskStatus, error) {
+	systemd.ApplyDaemonReload()
+	/*////////////////////////////////////////////////////
+	First thing to do is to check if the Name given is outside
+	the normal search path of systemd. If so it should be linked
+	*/ ////////////////////////////////////////////////////
+	dir, unitName := filepath.Split(t.Name)
+	var shouldBeLinked = false
```

Comment:
choose `var shouldBeLinked bool` or `shouldBeLinked := false`. They do the same thing.